### PR TITLE
Improve error message layout

### LINKED ISSUE #19037: Error message in Google Sign-In form needs better CSS
**Describe the bug**
The padding on the error message surfaced in the Google Sign-In setting form is a bit strange.  This probably applies to other setting forms as well but I haven't checked.

**To Reproduce**
1. Go to Admin -> Settings -> Authentication -> Sign In with Google
2. Type gibberish into the "Client ID" form and save

**Expected behavior**
Error message should probably have less padding on the left (to be in line with the other form elements), and more on the bottom so that it's not so close to the save button.

**Screenshots**
Screen Shot 2021-11-18 at 11 28 40 AM

**Information about your Metabase Installation:**
Current master, though the Client ID validation is added in https://github.com/metabase/metabase/pull/19036 which hasn't been merged yet, so you have to check out that commit to use these repro steps

**Severity**
Very minor

### PR DESCRIPTION
**Pull request by gusaiani**
Fixes #19037

## In Google Auth Settings Page

1. Go to http://localhost:3000/admin/settings/authentication/google
2. Type gibberish into the "Client ID" form and save

## When adding or editing a database connection

1. Go to http://localhost:3000/admin/databases/create
2. Type gibberish into the inputs and save

### CODE REVIEW

**Comment by kulyk** [comment]
The code looks great, though I'm not sure about the vertical margin here. I'd tag someone from the design team for a review

Here's how it looks now. The error message is much closer to the "Show advanced option" thing rather than to the bottom, so it's not clear where does the error belongs (if it makes sense?)

CleanShot 2022-01-20 at 16 50 55@2x

On the other hand, here's how it looked like in Metabase x.40 (was already broken on x.41):

CleanShot 2022-01-20 at 17 08 48@2x

**Comment by gusaiani** [comment]
@mazameli, wdyt of this positioning for the error message?

As @kulyk pointed out, we had it flushed right before. Then it broke.

My justification for simply placing it left-hand and above Save or Submit buttons is that they'll just be sturdier, less prone to layout problems and spooky side-effects.

**Comment by gusaiani** [comment]
@kulyk I just got word from Nathalia that a whole new set of form styles is soon to come: https://www.figma.com/file/UZVft5Kq9i7QijXGu5akQ2/Database-errors?node-id=1%3A14

We can wait for Maz to share his point of view, or maybe we can just merge this and know that something new is in store.

File: frontend/src/metabase/components/form/FormMessage.styled.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import styled, { css } from "styled-components";
+
+import { space } from "metabase/styled-components/theme";
+import { color } from "metabase/lib/colors";
+
+export const FormMessageStyled = styled.span`
+  color: ${({ hasSucceeded }) =>
+    hasSucceeded ? color("success") : color("error")};
+  float: left;
```
Comment (nemanjaglumac) [inline frontend/src/metabase/components/form/FormMessage.styled.jsx:9]: Why `float`?
I'm afraid it might come back and bite us.

This span is inside a `div` and that parent div is not cleared (it has a height of 0).

Comment (gusaiani) [reply frontend/src/metabase/components/form/FormMessage.styled.jsx:9]: @nemanjaglumac, I looked into this and came out thinking we're better off with a float than what we have or had.

See the visual example of the Google auth above.

The error message is very long, and with long copy, keeping it to the right as we had before is prone to strange or broken layouts.

Also, the wrapping div is a sign that this is fragile, right? To depend on people knowing to add a container element, and to treat them in a particular way.

It's a good thing we are developing proper components for this soon. 

To sum it up, the way we do it here seems to make the error messages just land above the submit buttons. Unless I'm missing something, I think it's an acceptable solution until we get the components.

Comment (nemanjaglumac) [reply frontend/src/metabase/components/form/FormMessage.styled.jsx:9]: If it's a temporary solution, so be it.
It currently looks better than before. I'm sure we could find some edge cases where `float` would backfire if we poke at it long enough. But it's pointless to invest much time in something that's not such a high priority.

**Review by nemanjaglumac — APPROVED** [review]
